src/dicomweb_client/api.py

```diff
@@ -2187,28 +2216,24 @@ def retrieve_instance(
             'wado', study_instance_uid, series_instance_uid, sop_instance_uid
         )
         if media_types is None:
-            return self._http_get_multipart_application_dicom(url)[0]
+            return list(self._http_get_multipart_application_dicom(url))[0]
         common_media_type = self._get_common_media_type(media_types)
         if common_media_type == 'application/dicom':
-            return self._http_get_multipart_application_dicom(
+            return list(self._http_get_multipart_application_dicom(
                 url, media_types
-            )[0]
+            ))[0]
         elif common_media_type == 'application/octet-stream':
-            return self._http_get_multipart_application_octet_stream(
+            return list(self._http_get_multipart_application_octet_stream(
                 url, media_types
-            )[0]
+            ))[0]
         elif common_media_type.startswith('image'):
-            frames = self._http_get_multipart_image(
-                url, media_types
-            )
+            frames = list(self._http_get_multipart_image(url, media_types))
             if len(frames) > 1:
                 return frames
```

Comment:
@dclunie @fedorov do you have any thoughts on this?

Reply:
That should not conflict with the Google Healthcare API (see [conformance statement](https://cloud.google.com/healthcare/docs/dicom))

Reply:
Yes, I would remove them to match the latest standard.  I don't think they have been widely used.

Reply:
I am not familiar with this, so I defer to Steve and David.

Reply:
I would be extremely cautious about what the current PS3.18 appears to have removed during the "re-documentation" in Sup 183 - we (DICOM WG 6 and WG 27) are actively trying to repair the damage done to the media type documentation, so things that appear to have been removed may come back (and things that may appear to have become permitted may not be). The intent of Sup 183 was to not change the behavior.

E.g., see in particular:

ftp://medical.nema.org/medical/dicom/cp/cp2040_01_Fix_DicomWeb_DICOM_Media_Types.pdf

Also, FYI ...

ftp://medical.nema.org/medical/dicom/cp/cp1614_05_Bulkdata_endpoint.pdf

Note in particular that the response is always some flavor of "multipart/related".

Reply:
Ah, good to know @dclunie, thank you.

Reply:
Thanks @dclunie for the clarification. The "re-documentation" efforts has indeed created a lot of confusion and apparently also broke things.

I always found it weird that `application/octet-stream`, `image/*` and `video/*` media types are allowed for the Retrieve Instance transaction, since the thereby retrieved bulkdata only represent part of the instance. Therefore, I agree with CP1614 and suggest implementing the retrieval of study, series and instance bulkdata via separate `retrieve_{study,series,instance}_bulkdata` methods. In other words, let's move this code for handling `application/octet-stream` media type from the `retrieve_{study,series,instance}` to the `retrieve_{study,series,instance}_bulkdata` method. 

Let's remove support for `image/*` and `video/*` media type until it is clear how "pixel data" is supposed to be handled. To me, `/pixels` doesn't make sense, since we already have `/frames`. I would rather suggest allowing `/frames` without requiring frame numbers.

Reply:
I gave this a shot based on the comments and the linked docs, but I'm not completely sure this was the goal. I'll try to add at least one happy path test and cross-check behavior with Orthanc, but any and all suggestions are welcome.

Reply:
@ambrussimon I have given this some further thought. Let's remove the `retrieve_*_bulkdata` methods for now until the expected behaviour is clarified in the standard. In the meantime, we will no longer support `image/*` and `video/*` or `application/octet-stream` media types for the `retrieve_*` methods, but only `application/dicom`. However, let's keep the `media_types` parameter.

Reply:
I am going to merge this into a feature branch